velox/common/base/CompareFlags.h

```diff
@@ -14,38 +14,137 @@
  * limitations under the License.
  */
 #include <fmt/core.h>
+#include <optional>
 #include <sstream>
 #include <string>
 
 #pragma once
 
 namespace facebook::velox {
 
+constexpr auto kIndeterminate = std::nullopt;
+
 // Describes value collation in comparison.
 struct CompareFlags {
-  enum class NullHandlingMode {
-    // This is the default null handling mode, in this mode nulls are treated as
-    // values such that:
-    //  - null == null is true,
-    //  - null == value is false.
-    //  - when equalsOnly=false null ordering is determined using the nullsFirst
-    // flag.
-    kNullAsValue,
-    kStopAtNull
-  };
-
   bool nullsFirst = true;
 
   bool ascending = true;
 
   // When true, comparison should return non-0 early when sizes mismatch.
```

Comment:
This comment suggests that equalsOnly = true is an optimization, but it is not. The semantics of equalsOnly = true and false are different. It would be nice to update the comment to avoid confusion.